connector/src/main/scala/com/microsoft/kusto/spark/datasink/CountingCsvWriter.scala

```diff
@@ -33,13 +33,40 @@ case class CountingCsvWriter(out: Writer) {
 
       bytesCounter += 2
 
-      for (c <- str) {
-        if (c == '"') {
-          out.write("\"\"")
-          bytesCounter += 1
+      // Json does not allow special characters
+      if(nested){
+        for (c <- str) {
+          if (c == '\n') {
+            out.write("\\n")
+            bytesCounter += 1
+          } else if ( c == '\t'){
+            out.write("\\t")
+            bytesCounter += 1
+          } else if (c == '\b'){
+            out.write("\\b")
+            bytesCounter += 1
+          } else if ( c == '\f') {
+            out.write("\\f")
+            bytesCounter += 1
+          } else if (c == '\t'){
+            out.write("\\t")
+            bytesCounter += 1
```

Comment:
you have \t twice.
maybe have a collection of escapable chars and iterate them?
```java 
if (nested) {
    for (c <- str) {
        for ( escapable <- ['\t', '\n'...] {
            if c == esacapeable {
                     out.write('\' + c);
             }
        }
    }
}
```

Reply:
This wont work as theres a differnce between
'\r' and '\\r'
One is a special char , the later is slash followed by the letter r

Reply:
This wont work as theres a differnce between
'\r' and '\\r'
One is a special char , the later is slash followed by the letter r